hydromt/models/model.py

```diff
@@ -246,32 +230,35 @@ def build(
                 }
 
         """
-        opt = opt or {}
-        opt = self._check_get_opt(opt)
-
-        # merge cli region and res arguments with opt
-        if region is not None:
-            if self._CLI_ARGS["region"] not in opt:
-                opt = {self._CLI_ARGS["region"]: {}, **opt}
-            opt[self._CLI_ARGS["region"]].update(region=region)
-
-        # then loop over other methods
-        for method in opt:
-            # if any write_* functions are present in opt, skip the final self.write()
-            if method.startswith("write_"):
-                write = False
-            kwargs = {} if opt[method] is None else opt[method]
-            self._run_log_method(method, **kwargs)
-
-        # write
-        if write:
+        steps = steps or {}
+        validate_steps(self, steps)
+
+        # steps gets preference over defaults.
+        # But put region.create at the start of the list.
+        steps = pydantic.v1.utils.deep_update(
+            steps, {"region.create": {"region": region}}
+        )
+
+        tmp = steps.pop("region.create")
+        steps = {"region.create": tmp, **steps}
```

Comment:
I don't think we want to enforce this. We had a discussion about region for the new issue so we can also change later but I also think some model like wflow will prefer to keep their setup_basemaps method which creates the region and also adds DEM data. This is because hydrography_fn (like merit_hydro) should really be the same both for region.create, grid.create and grid.add_topography_data so to avoid inconsistency I think wflow will still keep one method instead of splitting. Even if they do I think we should keep it flexible.

Reply:
Yes I agree that we don't want to enforce `region.create` in the front. Already working on this in a new branch.

Reply:
adressed by #847